Services/Masterdata/Services.Masterdata.EF/AutomapperProfile.cs

```diff
@@ -11,11 +14,92 @@ public AutomapperProfile()
         // Masterdata
         CreateMap<MasterdataType, MasterdataTypeListModel>();
         CreateMap<MasterdataType, MasterdataTypeModel>();
-        CreateMap<MasterdataTypeCreateModel, MasterdataType>();
-        CreateMap<MasterdataTypeUpdateModel, MasterdataType>();
+        CreateMap<MasterdataTypeCreateModel, MasterdataType>()
+            .AfterMap(UpdateMetadata);
+        CreateMap<MasterdataTypeUpdateModel, MasterdataType>()
+            .AfterMap(UpdateMetadata);
 
         CreateMap<Entities.Masterdata, MasterdataModel>();
         CreateMap<MasterdataCreateModel, Entities.Masterdata>();
         CreateMap<MasterdataUpdateModel, Entities.Masterdata>();
     }
+
+    private void UpdateMetadata(IMetadataModel source, MasterdataType destination)
+    {
+        // if nothing was passed in, or we have no clue on the domain, we do nothing with the metadata
+        if (source.Metadata is null || string.IsNullOrEmpty(source.Domain)) return;
+
+        Dictionary<string, dynamic>? finalMetadataDictionary = null;
+
+        // if the database already contains metadata
+        if (destination?.MetadataJson != null) 
+        {
+            finalMetadataDictionary = JsonSerializer.Deserialize<Dictionary<string, dynamic>>(destination.MetadataJson);
+            if (finalMetadataDictionary != null)
+            {
+                // We assume you passed in a complete metadata object (dictionary<string, dynamic)
+                // We merge this nicely with the given metadata
+                if (source.Domain == IMetadataModel.AllDomains)
+                {
+                    // for each passed in keyValue
+                    string sourceMetadataJson = source.Metadata.ToString();
+                    var sourceDictionary = JsonSerializer.Deserialize<Dictionary<string, dynamic>>(sourceMetadataJson);
+                    sourceDictionary.ForEach(keyValue =>
+                    {
+                        UpdateMetadataDictionary(keyValue.Key, keyValue.Value, finalMetadataDictionary);
+                    });
+                }
+                // an metadata object was passed in for a specific domain
+                // we only update that object
+                else
+                {
+                    UpdateMetadataDictionary(source.Domain, source.Metadata, finalMetadataDictionary);
+                }
+            }
+        }
+        else
+        {
+            if(source.Domain != IMetadataModel.AllDomains)
+            {
+                finalMetadataDictionary = new Dictionary<string, dynamic>() { { source.Domain, source.Metadata } };
+            }
+            else
+            {
+                finalMetadataDictionary = source.MetadataDictionary;
+            }
+        }
+
+        if(finalMetadataDictionary != null)
+            destination!.MetadataJson = JsonSerializer.Serialize(finalMetadataDictionary);
```

Comment:
dangerous notation here. Accolades indeed not needed in a single statement if, but as soon as a second line is added, you need to add them. Why not add them in advance in the context of defensive programming?